astropy/table/setup_package.py

```diff
@@ -1,22 +1,23 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-import os
+
+from pathlib import Path
 
 from numpy import get_include as get_numpy_include
 from setuptools import Extension
 
-ROOT = os.path.relpath(os.path.dirname(__file__))
+ROOT = Path(__file__).parent.relative_to(Path.cwd())
 
 
 def get_extensions():
-    sources = ["_np_utils.pyx", "_column_mixins.pyx"]
+    sources = [ROOT / "_np_utils.pyx", ROOT / "_column_mixins.pyx"]
     include_dirs = [get_numpy_include()]
 
     exts = [
         Extension(
-            name="astropy.table." + os.path.splitext(source)[0],
+            name=f"astropy.table.{source.stem}",
             define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
-            sources=[os.path.join(ROOT, source)],
+            sources=sources,
```

Comment:
Is it known that `Extension` can take a list of `Path` instead of a list of `str`? Does this code get tested when building/installing astropy in CI testing?

Reply:
It's been supported by setuptools since version 72.2.0, which is why I set an explicit minimum requirement.